dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,958 @@
+                    .append("\")");
+            }
+        } else {
+            sb.append("#set($isWidget= \"")
+                .append(false)
+                .append("\")");
+        }
+
+        // close contentlet div
+        sb.append("#if($API_EDIT_MODE)")
+            .append("</div>")
+            .append("#end");
+
+        return writeOutVelocity(filePath, sb.toString());
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<Contentlet> sortContentlets(List<Contentlet> contentletList, String sortBy) {
+
+        Logger.debug(this.getClass(), "I'm ordering by " + sortBy);
+
+        if (sortBy.equals("tree_order")) {
+            Collections.sort(contentletList, new WebAssetSortOrderComparator());
+        } else {
+            Collections.sort(contentletList, new ContentComparator("asc"));
+        }
+
+        return contentletList;
+
+    }
+
+    /**
+     * Will remove all contentlet files within a structure for both live and working. Uses the
+     * system user.
+     * 
+     * @param contentlets
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    public void removeContentletFile(Structure structure) throws DotDataException, DotSecurityException {
+        ContentletAPI conAPI = APILocator.getContentletAPI();
+        int limit = 500;
+        int offset = 0;
+        List<Contentlet> contentlets = conAPI.findByStructure(structure, APILocator.getUserAPI()
+            .getSystemUser(), false, limit, offset);
+        int size = contentlets.size();
+        while (size > 0) {
+            for (Contentlet contentlet : contentlets) {
+                invalidateAll(contentlet);
+            }
+            offset += limit;
+            contentlets = conAPI.findByStructure(structure, APILocator.getUserAPI()
+                .getSystemUser(), false, limit, offset);
+            size = contentlets.size();
+        }
+    }
+
+    private void removeContentletFile(Contentlet asset, Identifier identifier, boolean EDIT_MODE)
+            throws DotDataException, DotSecurityException {
+        CacheLocator.getContentletCache()
+            .remove(asset.getInode());
+
+        String folderPath = (!EDIT_MODE) ? "live" + java.io.File.separator : "working" + java.io.File.separator;
+        String velocityRootPath = VelocityUtil.getVelocityRootPath();
+        velocityRootPath += java.io.File.separator;
+
+        Set<Long> langs = new HashSet<Long>();
+        langs.add(asset.getLanguageId());
+        if (LanguageWebAPI.canApplyToAllLanguages(asset)) {
+            for (Language ll : APILocator.getLanguageAPI()
+                .getLanguages()) {
+                langs.add(ll.getId());
+            }
+        }
+        for (Long langId : langs) {
+            String filePath = folderPath + identifier.getId() + "_" + langId + "." + VelocityType.CONTENT.fileExtension;
+            java.io.File f = new java.io.File(velocityRootPath + filePath);
+            f.delete();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "java.nio.Files#delete" here for better messages on error conditions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Do something with the "boolean" value returned by "delete". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS899)